R/doeFactorial.R

```diff
@@ -175,100 +181,200 @@ doeFactorial <- function(jaspResults, dataset, options, ...) {
+                                repeat.only = rep,
                                 randomize = rnd)
-        rows <- data.frame(runOrder = runOrder)
-        rows <- cbind.data.frame(rows,
-                                 splitRuns)
       } else {
         splitRuns <- FrF2::FrF2(nfactors = options[["numberOfFactors"]],
                                 resolution = ifelse(options[["factorialResolution"]] != "Full",
                                                     as.numeric(as.roman(options[["factorialResolution"]])),
                                                     999),
                                 hard = options[["numberHTCFactors"]],
+                                replications = options[["factorialCornerReplicates"]],
+                                repeat.only = rep,
                                 randomize = rnd)
-        runOrder <- 1:nrow(splitRuns)
-        rows <- data.frame(runOrder = runOrder)
-        rows <- cbind.data.frame(rows,
-                                 splitRuns)
       }
-    } #blocks NO
+      runOrder <- 1:nrow(splitRuns)
+      rows <- cbind.data.frame(runOrder, splitRuns)
+
+      table$addFootnote(paste("Hard-to-change factors: ", paste(factorNames[1:options[["numberHTCFactors"]]], collapse = ", "), sep = ""))
+    }
 
+    #full
     if (options[["factorialType"]] == "factorialTypeFull") {
-      runOrder <- 1:2^options[["numberOfFactors"]]
-      rows <- data.frame(runOrder = runOrder)
-      rows <- cbind.data.frame(rows,
-                               FrF2::FrF2(nfactors = options[["numberOfFactors"]],
-                                          nruns = 2^options[["numberOfFactors"]],
-                                          randomize = rnd))
-    } #blocks YES
+      rows <- FrF2::FrF2(nfactors = options[["numberOfFactors"]],
+                         nruns = 2^options[["numberOfFactors"]],
+                         ncenter = options[["factorialCenterPoints"]],
+                         replications = options[["factorialCornerReplicates"]],
+                         repeat.only = rep,
+                         blocks = options[["factorialBlocks"]],
+                         randomize = rnd)
+      runOrder <- 1:nrow(rows)
+      rows <- cbind.data.frame(runOrder, rows)
+    }
+
+    #p-b
+    if(options[["factorialType"]] == "factorialPlackettBurman"){
+      rows <- FrF2::pb(nruns = as.numeric(options[["PBruns"]]),
+                       nfactors = options[["numberOfFactors"]],
+                       ncenter = options[["factorialCenterPoints"]],
+                       replications = options[["factorialCornerReplicates"]],
+                       repeat.only = rep,
+                       randomize = rnd)
+      runOrder <- 1:nrow(rows)
+      rows <- cbind.data.frame(runOrder, rows)
+    }
 
+    blocks <- rows$Blocks
+    rows <- rows[,!names(rows) %in% "Blocks"]
+
+    #filling in table
     for (i in 1:as.numeric(options[["numberOfFactors"]])) {
```

Comment:
So sometimes you do `as.numeric(options[["numberOfFactors"]])` and sometimes you don't. In the QML, I see that numberOfFactors is an integerfield so I think you can just always remove the `as.numeric`.

Reply:
It's tricky because there have been times where, say, a function wouldn't work unless the value from an IntegerField were wrapped in ```as.numeric```, so I just started wrapping many different values in it "out of safety". I'll check each instance of this manually, to see when ```as.numeric``` is really necessary .

Reply:
That's not a good approach. You should know more or less what values an object contains. If you need to cast something between different types, fine, but do it once (in the entire analysis) and then store it. Code generally becomes most clear when the assumptions are clear. If the assumptions are violated, it's fine to fail early.